Apps/TestProtocol/testBNProtocol.js

```diff
@@ -0,0 +1,130 @@
+                    execute(`mkdir build`, "./BabylonNative", (error, stdout, stderr) => {
+                        if (error) throw error;
+                        execute(`npm install`, "./BabylonNative/Apps", (error, stdout, stderr) => {
+                            if (error) throw error;
+                            console.log('Building win32 project.');
+
+                            execute(`cmake -G "Visual Studio 16 2019" -A x64 ..`, "./BabylonNative/build", (error, stdout, stderr) => {
+                                if (error) throw error;
+                                console.log('Building win32 apps.');
+                                execute(`cmake --build build --config Release`, "./BabylonNative", (error, stdout, stderr) => {
+                                    if (error) throw error;
+                                    patchTestScript();
+                                    callback(tag, hash);
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
+}
+
+function testPackages(tag, hash) {
+    let compatiblePackageVersions = [];
+    console.log("Getting NPM versions ...");
+    execute("npm show babylonjs@5.*.* version --json", "./", (error, stdout, stderr) => {
+        if (error) throw error;
+    
+        const versions = JSON.parse(stdout);
+    
+        versions.forEach(version => {
+            console.log("-".repeat(30));
+            console.log(`Installing packages for ${version}`);
+            execute(`npm install babylonjs@${version} babylonjs-gui@${version} babylonjs-loaders@${version} babylonjs-materials@${version}`, "./", (error, stdout, stderr) => {
+                if (error) throw error;
+                    
+                // copy package .js
+                console.log("Copying packages files.");
+                filesToCopy.forEach(entry => {
+                    entry.files.forEach(file => {
+                        copy(`node_modules/${entry.source}/${file}`, `${exeFolder}/Scripts/${file}`);
+                    });
+                });
+    
+                // run validation test
+                console.log("Running validation tests.");
+                execute("ValidationTests", exeFolder, (error, stdout, stderr) => {
+                    console.log(`Error Code ${error}`);
+                    if (!!error) {
+                        console.log(`${version} Failed!`);
+                    } else {
+                        compatiblePackageVersions.push(version);
+                        console.log(`${version} OK!`);
+                    }
+                });
+            });
+        });
+    });
+    console.log(`Compatible versions for BabylonReactNative ${tag} using BabylonNative ${hash}:`, compatiblePackageVersions);
+    compatibility.push({tag:tag, hash:hash, npms:compatiblePackageVersions});
+}
+
+BRNVersions.forEach((versionToTest) =>{
```

Comment:
Is it a goal to be able to run this for every version locally? Would it make more sense for the version to be an input to the script, and then use a GitHub actions matrix thing to be able to run them all in parallel across multiple machines?

Reply:
Yes, it would be faster to run. And it should also use the result from last run to not re-test everything.